### ISSUE TITLE
S3925 FP: should not raise on sealed exception 

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#2580]
### Description

If I mark an exception class with sealed, it will trigger.

### Repro steps

```
    [Serializable]
    public sealed class MySealedException : Exception // Noncompliant
    {
        public MySealedException () { }
        public MySealedException (string message) : base(message) { }
        public MySealedException (string message, Exception innerException) : base(message, innerException) { }
        protected MySealedException (SerializationInfo info, StreamingContext context) : base(info, context) { }
    }

    [Serializable]
    public class MyNotSealedException : Exception // Ok
    {
        public MySealedException () { }
        public MySealedException (string message) : base(message) { }
        public MySealedException (string message, Exception innerException) : base(message, innerException) { }
        protected MySealedException (SerializationInfo info, StreamingContext context) : base(info, context) { }
    }

```

### Expected behavior

If the exception is sealed or not, it should not trigger a warning.

### Actual behavior

Triggers on sealed exception class.

### Known workarounds

Don't mark it as sealed.

### Related information

* SonarC# Version 7.15

**Comment by andrei-epure-sonarsource** [comment]
Looking better at the rule description, if the `A sealed type has a serialization constructor that is not private.` - in the above reproducer, the constructor is not private...
